File: examples/javascript/constructs/functions_basic.js
// Ported from examples/lumen/constructs/functions_basic.lm by scripts/port_examples.py; edit the Lumen original, not this file.
function square(x) {
    return x * x;
}

function add(a, b) {
    return a + b;
}

function greet(name) {
    return "Hello, " + name;
}

console.log("Test: Basic Functions");
console.log(square(5));
console.log(add(10, 20));
console.log(greet("Lumen"));
function get_constant() {
    return 42;
}

console.log(get_constant());
function compute(x, y) {
    const sum = x + y;
    const product = x * y;
    return sum + product;
}

console.log(compute(3, 4));

File: examples/rust/constructs/functions_basic.rs
// Ported from examples/lumen/constructs/functions_basic.lm by scripts/port_examples.py; edit the Lumen original, not this file.
fn square(x: i64) -> i64 {
    return x * x;
}

fn add(a: i64, b: i64) -> i64 {
    return a + b;
}

fn greet(name: String) -> String {
    return "Hello, " + name;
}

fn get_constant() -> i64 {
    return 42;
}

fn compute(x: i64, y: i64) -> i64 {
    let sum = x + y;
    let product = x * y;
    return sum + product;
}

fn main() {
    println!("Test: Basic Functions");
    println!("{}", square(5));
    println!("{}", add(10, 20));
    println!("{}", greet("Lumen"));
    println!("{}", get_constant());
    println!("{}", compute(3, 4));
}
